tests/training/trainer_test.py

```diff
@@ -61,6 +61,25 @@ def test_trainer_can_resume_training(self):
         assert val_metrics_per_epoch[0] != 0.
         new_trainer.train()
 
+    def test_should_stop_early_positive(self):
+        new_trainer = Trainer(self.model, self.optimizer,
+                              self.iterator, self.dataset,
+                              validation_dataset=self.dataset,
+                              num_epochs=3, serialization_dir=self.TEST_DIR,
+                              patience=5, validation_metric="+test")
+        assert new_trainer._should_stop_early([.5, .3, .2, .1, .4, .4])
+        assert not new_trainer._should_stop_early([.3, .3, .1, .2, .5, .3])
```

Comment:
Can you switch the .1 and the last .3 here, so that we can isolate the problem?  In other words, the best way to fix an issue is to first write a test that fails with current logic, show that it fails, then fix the code, and show that fixing the code fixes the test.  I think the current logic will pass this test, so you haven't isolated the issue yet.

Reply:
Oh, I might have this backwards...

Reply:
The "positive" and "negative" names could be better: `test_should_stop_early_with_increasing_metric` would be immediately obvious, and I wouldn't have had to figure out which case you were testing.